### PR TITLE
Command cooldowns and ratelimit system + misc. improvements

### PR DESCRIPTION
**Pull request by TheLegendarySpark**
This is my first time contributing this year. Woohoo!! I could say that I look forward to contributing again. After a year since I began developing Essential, my custom admin system, it's been real quiet for me. I definitely imported several functions from it for performance improvements. Hope you like it! 😊 

### Additions
+ Added settings.CommandCooldowns
+ Newly improved rate limit for process
+ Added several service functions: GetProductInfo, GetGroupCreatorId, CheckAssetOwnership, CheckPassOwnership, & GetProductInfo
+ Added Admin.UpdateCooldown

### Improvements & Changes
+ Admin.DoCheck no longer requires a player object for checking gamepass ownership and group ranks
   - This change reduces the number of yields for permission checking. It's also beneficial in the event of Roblox broken and other Roblox issues
+ Reorganized the code inside Process.Command and moved the permission checks to Admin.CheckPermission

### CODE REVIEW

**Comment by P3tray** [comment]
Welcome back.

**Comment by Sceleratis** [comment]
Please do not use dumb pull titles qq

File: MainModule/Server/Core/Admin.lua [hunk 1]
```diff
@@ -224,7 +240,7 @@ return function(Vargs, GetEnv)
 		--// How long admin levels will be cached (unless forcibly updated via something like :admin user)
 		AdminLevelCacheTimeout = 30;
 
-		DoHideChatCmd = function(p: Player, message: string, data: {[string]: any}?)
+		DoHideChatCmd = function(p, message, data)
```
Comment (Sceleratis) [inline MainModule/Server/Core/Admin.lua:243]: Why'd you remove the types?

**Review by P3tray — CHANGES_REQUESTED** [review]
@TheLegendarySpark This pull is unsuitable in it's current state. It reverts changes done by @Expertcoderz previously, namely the type safety and index variable renaming. Can you please move your code to the latest branch?

**Comment by Expertcoderz** [comment]
Merged, now time to test if everything in this huge update works as intended.

**Pull request merged**